src/critnib/critnib.c

```diff
@@ -410,16 +462,34 @@ void *critnib_remove(struct critnib *c, word key) {
         goto not_found;
     }
 
-    word del =
-        (utils_atomic_increment_u64(&c->remove_count) - 1) % DELETED_LIFE;
+    word del;
+    int i_del = 0;
+    uint64_t ref_count = 0;
+    // We look for a leaf with ref_count == 0.
+    // If we do not find one we take the first one that was checked.
+    // In such case (extremely unlikely) we might not be able to free the value
+    // and a memory leak might happen.
```

Comment:
add a log message in this case

Reply:
It is verified and logged in `alloc_leaf()`.